src/item.cpp

```diff
@@ -2373,7 +2377,7 @@ std::string item::display_name(unsigned int quantity) const
         qty = string_format(" (%i)", ammo_remaining());
     } else if( is_ammo_container() && !contents.empty() ) {
         qty = string_format( " (%i)", contents.front().charges );
-    } else if( count_by_charges() ) {
+    } else if( charges > 0 || count_by_charges() ) {
```

Comment:
Are you trying to enable a countdown-type thing? There shouldn't be one. Dynamite is kind of weird in that it's visible, but the countdown state of other things shouldn't be, and in fact could potentially be a little randomized.